### PR TITLE
User Sign Up Status

### PR DESCRIPTION
**Pull request by AKHIL-882**
The endpoint had created to know the status of the user during the signup process.

The endpoint determines the users to know in which steps the process had ended and need to start to create an account.

### CODE REVIEW

File: CoinDock-api/app/Http/Controllers/V1/Auth/UserController.php [hunk 1]
```diff
@@ -7,13 +7,46 @@
...
 use App\Models\V1\User;
+use App\Models\V1\SignUp;
 use Illuminate\Support\Facades\Auth;
 
 class UserController extends Controller
 {
     /**
      * SignupRequest
      */
+    public function signUpInfo(User $user)
```
Comment (sriragnair) [inline CoinDock-api/app/Http/Controllers/V1/Auth/UserController.php:23]: Should be in its own controller SignupController.
Return user specific data using the show() method

File: CoinDock-api/app/Http/Controllers/V1/Auth/UserController.php [hunk 2]
```diff
@@ -7,13 +7,46 @@
...
+    {
+        $signUp = SignUp::find($user->id);
```
Comment (sriragnair) [inline CoinDock-api/app/Http/Controllers/V1/Auth/UserController.php:25]: Signup is a single word.

File: CoinDock-api/app/Models/V1/SignUp.php [hunk 3]
```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace App\Models\v1;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class SignUp extends Model
```
Comment (sriragnair) [inline CoinDock-api/app/Models/V1/SignUp.php:8]: Same. Signup should be a single word

File: CoinDock-api/routes/v1/users.php [hunk 4]
```diff
@@ -23,6 +24,11 @@
     Route::get('logout', [UserController::class, 'logout']);
 });
 
-Route::get('/', [RecoveryKeyController::class, 'recoveyKey']);
+Route::get('/', [RecoveryKeyController::class, 'recoveryKey']);
 
 Route::get('/random' , [RecoveryKeyController::class, 'random']);
+
+Route::get('/users/{user}/signup/status/',[UserController::class,'signUpInfo'])
```
Comment (sriragnair) [inline CoinDock-api/routes/v1/users.php]: route should be 
api/user/{userId}/signup
according to REST api standard.

When looking for status: the endpoint should return only a status property. Here we are returning a signup object representing the state of the user signup

File: CoinDock-api/app/Http/Controllers/V1/Auth/UserController.php [hunk 5]
```diff
@@ -7,13 +7,46 @@
...
+
+        $resultArr = [
+            'step_1_completed' => false,
+            'step_2_completed' => false,
+            'step_3_completed' => false
+        ];
+
+        $stepCount = $signUp->step_count;
```
Comment (sriragnair) [inline CoinDock-api/app/Http/Controllers/V1/Auth/UserController.php:33]: What happens if $signup is null?

**Review by sriragnair — CHANGES_REQUESTED** [review]

File: CoinDock-api/database/migrations/2022_05_30_185417_create_signups_table.php [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    /**
+     * Reverse the migrations.
+     *
+     * @return void
+     */
+    public function down()
+    {
+        Schema::dropIfExists('user_statuses');
```
Comment (SwethaMattapalli) [inline CoinDock-api/database/migrations/2022_05_30_185417_create_signups_table.php]: Creating singups table and in drop method dropping user_statuses table?

File: CoinDock-api/app/Http/Controllers/V1/Auth/UserController.php (hunk 1 above)
Comment (AKHIL-882) [reply CoinDock-api/app/Http/Controllers/V1/Auth/UserController.php:23]: Modified

File: CoinDock-api/app/Http/Controllers/V1/Auth/UserController.php (hunk 2 above)
Comment (AKHIL-882) [reply CoinDock-api/app/Http/Controllers/V1/Auth/UserController.php:25]: Modified

**Comment by AKHIL-882** [comment]
Chances are made accordingly and updated the code in 1. Changes branch.

**Pull request closed without merging**